ddp-workspace/projects/ddp-atcp/src/app/components/header/header.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import {
+  CompositeDisposable,
+  ConfigurationService,
+  NGXTranslateService,
+  SessionMementoService,
+  UserProfileDecorator,
+  CurrentActivityService,
+  ActivityCodes
+} from 'ddp-sdk';
+import * as RouterResource from '../../router-resources';
+import { Language, LanguagesToken } from '../../providers/languages.provider';
+import { TranslateService } from '@ngx-translate/core';
+import { UserPreferencesServiceAgent } from '../../services/serviceAgents/userPreferencesServiceAgent';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.html',
+  styleUrls: ['./header.scss']
+})
+export class HeaderComponent implements OnInit, OnDestroy {
+  public RouterResource = RouterResource;
+  public SignIn: string;
+  private anchor = new CompositeDisposable();
+  private userProfileDecorator: UserProfileDecorator;
+  isProgressBarVisible = false;
+  activityToShowProgress = ActivityCodes.MEDICAL_HISTORY;
+
+  constructor(@Inject(LanguagesToken) public languages: Language[],
+              private session: SessionMementoService,
+              private ngxTranslate: NGXTranslateService,
+              private translate: TranslateService,
+              private currentActivityService: CurrentActivityService,
+              @Inject('ddp.config') private configuration: ConfigurationService,
+              private userPreferencesServiceAgent: UserPreferencesServiceAgent) {
+  }
+
+  public ngOnInit(): void {
+    const translate$ = this.ngxTranslate.getTranslation('Common.Navigation.SignIn')
+      .subscribe((SignIn: string) => this.SignIn = SignIn);
+    this.anchor.addNew(translate$);
+
+    const userProfileSubs$ = this.userPreferencesServiceAgent.profile
+      .subscribe((data: UserProfileDecorator) => {
+        if (!!data && this.isAuthenticated) {
+          this.userProfileDecorator = data;
+          this.runTranslator(this.userProfileDecorator.profile.preferredLanguage);
+        }
+      });
+    this.anchor.addNew(userProfileSubs$);
+
+    const currentActivity$ = this.currentActivityService.getCurrentActivity().subscribe(x => {
+      x && x.activityCode === this.activityToShowProgress
+      ? this.isProgressBarVisible = true
+        : this.isProgressBarVisible = false;
+      });
+    this.anchor.addNew(currentActivity$);
+  }
+
+  public get isAuthenticated(): boolean {
+    return this.session.isAuthenticatedSession();
+  }
+
+  public trackByFn(indx: number): number {
+    return indx;
+  }
+
+  public ngOnDestroy(): void {
+    this.anchor.removeAll();
+  }
+
+  public changeLanguage(language: Language): void {
+    if (!!this.userProfileDecorator) {
```

Comment:
We have a comment from 16th April:

> Re: application translation
> We use @ngx-translate in pepper. That means:
> 
> We store translation in https://github.com/broadinstitute/ddp-angular/blob/develop/ddp-workspace/projects/ddp-osteo/src/assets/i18n/en.json (osteo app for example)
> HTML uses translate pipe and keys from translation file https://github.com/broadinstitute/ddp-angular/blob/develop/ddp-workspace/projects/ddp-osteo/src/app/components/welcome/welcome.component.html (osteo app for example)
> Right now we don't have multi-language app, the SDK also doesn't have a component for switching language, this work remains to be done.

Should we create a special component for switching language ?

Reply:
Currently, ddp-language-selector can't be used in ATCP project (study doesn't have any languages configured).
Can we consider usage of ddp-language-selector a little bit later?
(Now we need an ability to test already translated parts).